spec/migrations/fix_publication_doi_fields_spec.rb

```diff
@@ -0,0 +1,42 @@
+require File.join(Rails.root, 'db', 'migrate', '20210610163713_fix_publication_doi_fields')
+
+describe 'Migration tests for Publication DOIs fix' do
+  let(:publication_doi) do
+    {
+      'MissingReason' => 'Unknown',
+      'Explanation' => 'explanation'
+    }
+  end
+
+  let(:available_doi) do
+    {
+      'DOI' => 'Publication reference DOI',
+      'Authority' => 'Publication reference authority'
+    }
+  end
+
+  context 'when doing the migration in MMT' do
+    before do
+      @draft = create(:full_collection_draft)
+      @template = create(:full_collection_template)
+
+      @draft.draft['PublicationReferences'][1]['DOI'] = publication_doi
+      @template.draft['PublicationReferences'][1]['DOI'] = publication_doi
+
+      @draft.save
+      @template.save
+
+      FixPublicationDoiFields.new.change
+    end
+
+    after do
+      CollectionDraft.delete([@draft.id])
+      CollectionTemplate.delete([@template.id])
+    end
+
+    it 'removes the DOI field from the second Publication Reference but not the first' do
+      expect(CollectionDraft.find(@draft.id).draft.dig('PublicationReferences',0,'DOI')).to eq(available_doi)
```

Comment:
Shouldn't it be removed from both the template and the draft? Why is it only removed from the template and not the draft? shouldnt 1 on both be nil?
Should we be checking that 0 on both is as it was before and 1 on both is now nil?